packages/cli/test/integration/server-type-files.test.ts

```diff
@@ -0,0 +1,108 @@
+import type express from 'express';
+import { mock } from 'jest-mock-extended';
+
+import type { AuthService } from '@/auth/auth.service';
+
+describe('Server Type Files Protection', () => {
+	let mockApp: jest.Mocked<express.Application>;
+	let mockAuthService: jest.Mocked<AuthService>;
+
+	beforeEach(() => {
+		mockApp = {
+			get: jest.fn(),
+			use: jest.fn(),
+		} as unknown as jest.Mocked<express.Application>;
+
+		mockAuthService = mock<AuthService>();
+		mockAuthService.createAuthMiddleware.mockReturnValue(jest.fn() as any);
+	});
+
+	describe('protected type files configuration', () => {
+		it('should register protected endpoints for type files with authentication', () => {
+			const staticCacheDir = '/mock/static/dir';
+
+			// Simulate the server configuration code
```

Comment:
If the implementation in the server changes, than this code would have to be kept in sync. Maybe it would be better to move this setup code into a function and reuse the function in the test as well?